search-commons/src/test/java/no/unit/nva/search2/OpenSearchAwsClientTest.java

```diff
@@ -0,0 +1,116 @@
+package no.unit.nva.search2;
+
+import static no.unit.nva.indexing.testutils.MockedJwtProvider.setupMockedCachedJwtProvider;
+import static no.unit.nva.indexing.testutils.SearchResponseUtil.getSearchResponseFromJson;
+import static no.unit.nva.search2.model.ResourceParameterKey.FROM;
+import static no.unit.nva.search2.model.ResourceParameterKey.SIZE;
+import static no.unit.nva.search2.model.ResourceParameterKey.SORT;
+import static nva.commons.core.ioutils.IoUtils.stringFromResources;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+import no.unit.nva.search2.model.OpenSearchQuery;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadRequestException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.client.RestHighLevelClient;
+
+class OpenSearchAwsClientTest {
+
+    private OpenSearchAwsClient openSearchAwsClient;
+
+    public static final String SAMPLE_OPENSEARCH_RESPONSE_RESPONSE_EXPORT
+        = "sample_opensearch_response.json";
+
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        var mockedOpenSearchClient = mock(RestHighLevelClient.class);
+        var mockedResponse = generateMockSearchResponse(SAMPLE_OPENSEARCH_RESPONSE_RESPONSE_EXPORT);
+        openSearchAwsClient = new OpenSearchAwsClient(setupMockedCachedJwtProvider(), mockedOpenSearchClient);
+        when(mockedOpenSearchClient.search(any(), any()))
+            .thenReturn(mockedResponse);
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("uriProvider")
+    void searchSingleTermReturnsOpenSearchAwsResponse(URI uri) throws ApiGatewayException {
```

Comment:
are all uri's single term?

Reply:
They used to be single term... I will update testNames